### ISSUE TITLE
Safe spots are due to crude line of sight in server

### ISSUE THREAD

**Issue by Meridian59** [Meridian59/Meridian59#23]
The server uses a very coarse representation of the map, which means that players can move/attack across boundaries that monsters cannot.  The server should load the full BSP tree from a roo file and use the same logic as the client for determining line of sight.

**Comment by cyberjunk** [comment]
The 103 branch has eliminated safe-spots.. while many like it, also many totally hated it :-)

I think we should have some serious discussion here whether this is an option for the vanilla branch at all or not. I'm especially interested in Zaphod's oppinion, because he opened this issue here. Would you really 'remove' safe-spots on your branch?

The technical implementation of the 103 approach to get rid of them is a bit different (less work to implement, but also less powerful) than suggestion here using BSP.

It's documented a bit on this wiki article:
http://wiki.openmeridian.org/index.php/High_Resolution_Grid,_LoS_and_monster_movement_upgrade

Basically we just added another server grid with an higher resolution than the old ones. This not only gets rid of safe-spots, but also allows monsters to step 'smoother' (smaller but more often). Since we store the height of the square too, we rather call it heightmap and it also allows us to do basic 3D distance calculations between objects in blakserv now (instead of 2D only before).

We also improved the serverside LoS, but making this correct would definitely require the BSP (and more computation...)

**Comment by cyberjunk** [comment]
Loading and using the BSP tree on the serverside has been added to the OpenMeridian branch.
If interested, see: https://github.com/OpenMeridian/Meridian59/pull/1079

**Comment by jrotunda** [comment]
We should look at the fix for lag drift to make combat feel better first. 

This fix would make Meridian much harder but it has been used for years now and players have shown they can deal with the adjustments. 

The backlash and effect here would be similar to the now fixed aggro check limit, as this is a similar bug introduced in the Revelations update many years ago - but I think the effect on gameplay would even out the difficulty and reduce the effectiveness of players who like to build with 90 mules and ruin the entire balance of the server every time they appear.
